frontend/src/pages/projects/screens/spawner/storage/WorkbenchStorageModal.tsx

```diff
@@ -0,0 +1,25 @@
+import * as React from 'react';
+import { PersistentVolumeClaimKind } from '~/k8sTypes';
+import { StorageData } from '~/pages/projects/types';
+import BaseStorageModal from '~/pages/projects/screens/detail/storage/BaseStorageModal';
+
+type WorkbenchStorageModalProps = {
+  existingData?: PersistentVolumeClaimKind;
+  onClose: (submit: boolean, storageData?: StorageData['creating']) => void;
+};
+
+const WorkbenchStorageModal: React.FC<WorkbenchStorageModalProps> = (props) => {
```

Comment:
Is this component meant to be used for editing cluster storage as well?

If so, looks like the title, submit label, etc can't be updated from external props.